src/cromshell/utilities/io_utils.py

```diff
@@ -42,3 +54,72 @@ def doomed_logo():
     print(colored(r"=='    _-'                                                            \/   `==", "red"))
     print(colored(r"\   _-'                                                                `-_   /", "red"))
     print(colored(r".`''                                                                      ``'.", "red"))
+
+
+def assert_required_file_is_not_empty(file_name: str, file_description: str):
+    """Confirm the provided file exist and is not empty."""
+
+    if file_name is None:
+        LOGGER.info("%s was not provided", file_description)
+        return
+    if not Path(file_name).exists():
+        LOGGER.error("ERROR: %s does not exist: %s", file_description, file_name)
+        raise Exception("ERROR: %s does not exist: %s", file_description, file_name)
+    else:
+        if 0 == os.stat(file_name).st_size:
+            LOGGER.error("ERROR: %s is empty: %s.", file_description, file_name)
+            raise Exception("ERROR: %s is empty: %s.", file_description, file_name)
+
+
+def validate_workflow_id(workflow_id: str):
+    """Validates a workflow id"""
+
+    pattern = re.compile("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}")
```

Comment:
This totally doesn't matter all in this case, but often you would precompile the expression outside of the function so it isn't recompiled on every invocation.  It's a performance optimization that can be large if the function is called often.

Reply:
done